docusaurus/docs/protocol/upgrades/contigency_plans.md

```diff
@@ -0,0 +1,66 @@
+---
+title: Failed upgrade contingency plan
+sidebar_position: 5
+---
+
+:::tip
+
+This documentation covers failed upgrade contingency for `poktroll` - a `cosmos-sdk` based chain.
+
+While this can be helpful for other blockchain networks, it is not guaranteed to work for other chains.
+
+:::
+
+## Contingency plans <!-- omit in toc -->
+
+There's always a chance the upgrade will fail.
+
+This document is intended to help you recover without significant downtime.
+
+- [Option 0: The bug is discovered before the upgrade height is reached](#option-0-the-bug-is-discovered-before-the-upgrade-height-is-reached)
+- [Option 1: The upgrade height is reached and the migration didn't start](#option-1-the-upgrade-height-is-reached-and-the-migration-didnt-start)
+- [Option 2: The migration is stuck](#option-2-the-migration-is-stuck)
+- [Option 3: The network is stuck at the future height after the upgrade](#option-3-the-network-is-stuck-at-the-future-height-after-the-upgrade)
+
+### Option 0: The bug is discovered before the upgrade height is reached
+
+**Cancel the upgrade plan!!**
+
+See the instructions of [how to do that here](./upgrade_procedure.md#cancelling-the-upgrade-plan).
+
+### Option 1: The upgrade height is reached and the migration didn't start
+
+If the nodes on the network stopped at the upgrade height and the migration did not
+start yet (i.e. there are no logs indicating the upgrade handler and store migrations are being executed),
+we mist gather social consensus to restart validators with the `--unsafe-skip-upgrade=$upgradeHeightNumber` flag.
+
+This will skip the upgrade process, allowing the chain to continue and the protocol team to plan another release.
+
+`--unsafe-skip-upgrade` simply skips the upgrade handler and store migrations.
+The chain continues as if the upgrade plan was never set.
+The upgrade needs to be fixed, and then a new plan needs to be submitted to the network.
+
+:::caution
+
+`--unsafe-skip-upgrade` needs to be documented and added to the scripts so the next time somebody tries to sync the network from genesis - they will automatically skip the failed upgrade.
+
+<!-- TODO_IMPROVE(@okdas): new cosmovisor UX can simplify this -->
+
+:::
+
+### Option 2: The migration is stuck
+
+If the migration is stuck, there's always a chance the state has been mutated for
+the upgrade but the migration didn't complete.
+
+In such a case, we need to:
+
+- Roll back validators to the backup (a snapshot is taken by `cosmovisor` automatically prior to upgrade, if `UNSAFE_SKIP_BACKUP` is set to `false`).
```

Comment:
Please link to the appropriate doc/section for this

Reply:
Added a link to cosmovisor docs.